odoo/tools/sql.py

```diff
@@ -421,3 +420,19 @@ def pattern_to_translated_trigram_pattern(pattern):
 
     # replace the original wildcard characters by %
     return f"%{'%'.join(wildcard_escaped)}%" if wildcard_escaped else "%"
+
+def get_unique_psql_identifier(identifier: str) -> str:
+    """ If necessary, return truncated PostgreSQL ``identifier`` and enforce uniqueness.
+
+    :param identifier: identifier to be truncate uniquely
+    :return: unique identifier
+    """
```

Comment:
Using `get_X` is generally for accessors.  This makes a suitable identifier
```suggestion
def makel_identifier(identifier: str) -> str:
    """ Return ``identifier``, possibly modified to fit PostgreSQL's identifier size limitation.
    If too long, ``identifier`` is truncated and concatenated to some hash to make it mostly unique.
    """
```